### PR TITLE
Eli's impl

### PR DESCRIPTION
**Pull request by etkramer**

### CODE REVIEW

File: elijah/app.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+const getIsPostVisible = (postId: string, username: string) => {
+  const post = getPostById(postId);
+  const author = getUserByUsername(post.userId);
+  const user = getUserByUsername(username);
+
+  let hasAccess = post.visibility === "public";
+  hasAccess = hasAccess || author.username === user.username;
+  hasAccess = hasAccess || author.friends.includes(user.username);
+
+  return hasAccess;
```
Comment (rchagolla) [inline elijah/app.ts:42]: Lines 33-35 are a very neat way to check post access of a user.

File: elijah/app.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    const choice = await rl.question(`
+1. Load input data.
+2. Check visibility.
+3. Retrieve posts.
+4. Search users by location.
+5. Exit.
+Choice: `);
+
+    switch (choice) {
+      case "1": {
```
Comment (rchagolla) [inline elijah/app.ts:143]: Move everything in case 1 into a separate function.

File: elijah/app.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+type Post = {
+  postId: string;
+  userId: string;
+  visibility: "public" | "friend";
+};
+
+let posts: Post[] = [];
+let users: User[] = [];
+
+const getPostById = (id: string) =>
```
Comment (rchagolla) [inline elijah/app.ts]: All get helper functions are clean, logical, and concise.

File: elijah/app.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  while (shouldContinue) {
+    const choice = await rl.question(`
+1. Load input data.
+2. Check visibility.
+3. Retrieve posts.
+4. Search users by location.
+5. Exit.
+Choice: `);
+
+    switch (choice) {
```
Comment (rchagolla) [inline elijah/app.ts]: Great use of switch case statement.

File: elijah/app.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+};
+
+type Post = {
+  postId: string;
+  userId: string;
+  visibility: "public" | "friend";
+};
+
+let posts: Post[] = [];
+let users: User[] = [];
```
Comment (Gabriel3970) [inline elijah/app.ts:23]: Good way to keep track of all different Posts and Users created. Had trouble with this in my code, will implement a similar system of User and Post arrays into mine.

File: elijah/app.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+4. Search users by location.
+5. Exit.
+Choice: `);
+
+    switch (choice) {
+      case "1": {
+        let postInfoPath = await rl.question(
+          "Path to post-info.txt (data/post-info.txt): "
+        );
+        let userInfoPath = await rl.question(
```
Comment (Gabriel3970) [inline elijah/app.ts]: Interesting way to loop the UI, I enjoy how clean typescript makes your code read.

File: elijah/app.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        if (!userInfoPath) {
+          userInfoPath = "data/user-info.txt";
+        }
+
+        const postInfoFile = (await readFile(postInfoPath)).toString();
+        const userInfoFile = (await readFile(userInfoPath)).toString();
+
+        posts = postInfoFile.split("\n").map((line) => {
+          const [postId, userId, visibility] = line.split(";");
```
Comment (Gabriel3970) [inline elijah/app.ts]: Handled storing postId, userId and visibility all in one line. Well done!

File: elijah/app.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+            state,
+            friends: friendsList.split("[")[1].split("]")[0].split(","),
+          };
+
+          return user;
+        });
+
+        break;
+      }
+      case "2": {
```
Comment (Gabriel3970) [inline elijah/app.ts:147]: Should have used switch statements in my code rather than if/else.

**Pull request closed without merging**